specification/taf-spec.md

```diff
@@ -0,0 +1,915 @@
+possibilities:
+
+- The change can be recorded in the authentication repository. In this case, the updater tool is
+designed to raise an error if this commit is not detected when cloning or updating `repo1`.
+- Alternatively, this commit can be omitted from being recorded in the authentication repository,
+in which case the updater will not look for it while traversing through commits of `repo1`.
+
+However, for repositories that are not permitted to contain unauthenticated commits, any new
+commit must be registered in the authentication repository. For instance, in the case of `repo1`,
+if it's not allowed to have unauthenticated commits, then its corresponding target file,
+`namespace1/repo1`, must be updated in the authentication repository to include the new commit.
+If the new commit in `repo1` is, for example, `e3f4g5h6`, then the next revision of the
+authentication repository will reflect this update:
+
+```markdown
+- metadata/
+  ├─ root.json
+  ├─ targets.json
+  ├─ snapshot.json
+  └─ timestamp.json
+- targets/
+  ├─ repositories.json
+  └─ mirrors.json
+  └─ namespace1/
+      ├─ repo1
+      │   └─ {"branch": "main", "commit": "e3f4g5h6"}
+      ├─ repo2
+      │   └─ {"branch": "main", "commit": "e5f6g7h8"}
+      └─ repo3
+          └─ {"branch": "main", "commit": "i9j0k1l2"}
+```
+
+Next, let's say that `branch1` is created in repositories `repo2` and `repo3` and one commit is
+created in each repository. Let's say that the two repositories are updated at the same time
+(meaning that if something is committed to `branch1` of `repo2`, something has to be committed to
+`branch1` of `repo3` as well). The current state of these target repositories would now look like
+this:
+
+```markdown
+namespace1/
+│
+├── repo1/
+│   └── main: a1b2c3d4, e3f4g5h6
+│
+├── repo2/
+│   ├── main: e5f6g7h8
+│   └── branch1: i7j8k9l0
+│
+└── repo3/
+    ├── main: i9j0k1l2
+    └── branch1: m2n3o4p5
+```
+
+The next revision of the authentication repository is then:
+
+```markdown
+- metadata/
+  ├─ root.json
+  ├─ targets.json
+  ├─ snapshot.json
+  └─ timestamp.json
+- targets/
+  ├─ repositories.json
+  └─ mirrors.json
+  └─ namespace1/
+      ├─ repo1
+      │   └─ {"branch": "main", "commit": "e3f4g5h6"}
+      ├─ repo2
+      │   └─ {"branch": "branch1", "commit": "e5f6g7h8"}
+      └─ repo3
+          └─ {"branch": "branch1", "commit": "m2n3o4p5"}
+```
+
+By navigating through the commit history of the authentication repository, it's possible to
+collect all registered commits for each target repository and each of their branches. So, within
+the TAF framework, there's no need to track all past commits of target repositories in a single
+revision of the authentication repository. Similarly, previous versions of metadata files can be
+easily accessed through the Git history.
+
+## 5 Updater
+
+This section focuses on the Updater, a key component of the TAF framework. The Updater's main job is
```

Comment:
We sometimes use Updater with capital U, but most of the time I've seen the updater in small case. Could we standardize?